File: server/src/models/course.py
from __future__ import annotations

from sqlmodel import SQLModel
from typing import Optional

from src.models.audit_model import AuditModel

class CourseBase(AuditModel):
    name: str = None
    course_code: str = None
    description: Optional[str] = None
    credits: Optional[float] = None

File: client/src/app/(course)/course.ts
import { AuditModel } from "../(students)/students";

type Course = {
    id: number;
    name: string;
    course_code: string;
    description: string | null;
    credits: number | null;
} & AuditModel;

type CourseActionParams = {
    courseId: number;
};

export type { Course, CourseActionParams }